### PR TITLE
[dbus] add capabilities API

### PR DESCRIPTION
**Pull request by erjiaqing**
This PR allows DBus client query the enabled build time features on the server.

Currently, nat64 is added to the list, more items can be added when needed.

### CODE REVIEW

File: src/dbus/server/introspect.xml [hunk 1]
```diff
@@ -929,6 +929,12 @@
       <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="false"/>
     </property>
 
+    <!-- TelemetryData: The Thread telemetry data (defined as proto/capabilities.proto)
+      in binary form. -->
```
Comment (superwhd) [inline src/dbus/server/introspect.xml:933]:
```suggestion
    <!-- Capabilities: The Thread capabilities (defined as proto/capabilities.proto)
      in binary form. -->
```

File: src/dbus/server/dbus_thread_object.cpp [hunk 2]
```diff
@@ -1523,6 +1526,24 @@ otError DBusThreadObject::GetTelemetryDataHandler(DBusMessageIter &aIter)
...
 
+otError DBusThreadObject::GetCapabilitiesHandler(DBusMessageIter &aIter)
+{
+    otError            error = OT_ERROR_NONE;
+    otbr::Capabilities capabilities;
+
+    capabilities.set_nat64(OTBR_ENABLE_NAT64);
+
+    {
+        const std::string    telemetryDataBytes = capabilities.SerializeAsString();
```
Comment (superwhd) [inline src/dbus/server/dbus_thread_object.cpp:1537]: Should be named as `capabilitiesBytes`?

File: src/dbus/client/thread_api_dbus.hpp [hunk 3]
```diff
@@ -869,6 +869,19 @@ class ThreadApiDBus
      */
     ClientError GetTelemetryData(std::vector<uint8_t> &aTelemetryData);
 
+    /**
+     * This method gets the telemetry data proto serialized byte data.
```
Comment (superwhd) [inline src/dbus/client/thread_api_dbus.hpp:873]:
```suggestion
     * This method gets the capabilities proto serialized byte data.
```

File: src/dbus/server/introspect.xml (hunk 1 above)
Comment (erjiaqing) [reply src/dbus/server/introspect.xml]: done

File: src/dbus/server/introspect.xml [hunk 4]
```diff
@@ -929,6 +929,12 @@
       <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="false"/>
     </property>
 
+    <!-- TelemetryData: The Thread capabilities data (defined as proto/capabilities.proto)
```
Comment (superwhd) [inline src/dbus/server/introspect.xml:932]:
```suggestion
    <!-- Capabilities: The Thread capabilities data (defined as proto/capabilities.proto)
```

**Review by superwhd — APPROVED** [review]
LGTM overall

**Review by superwhd — APPROVED** [review]
LGTM overall

**Review by superwhd — APPROVED** [review]
LGTM overall

**Review by xuyirio — APPROVED** [review]

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -31,8 +31,8 @@ if(OTBR_BORDER_AGENT)
     add_subdirectory(border_agent)
 endif()
 add_subdirectory(common)
-if(OTBR_FEATURE_FLAGS OR OTBR_TELEMETRY_DATA_API)
-    add_subdirectory(proto)
+if(OTBR_DBUS OR OTBR_FEATURE_FLAGS OR OTBR_TELEMETRY_DATA_API)
+add_subdirectory(proto)
```
[the remark this answers is not in the document]
Comment (erjiaqing) [reply src/CMakeLists.txt]: Removed the indent when removing the outter if, which breaks the build.

Reverted this line.

File: src/proto/capabilities.proto [hunk 6]
```diff
@@ -0,0 +1,13 @@
+syntax = "proto2";
+option optimize_for = LITE_RUNTIME;
+
+package otbr;
+
+// Capabilities message exposes a list of values of the macros of ot-br-posix. This is not an exhausted
```
Comment (jwhui) [inline src/proto/capabilities.proto:6]:
```suggestion
// Capabilities message exposes a list of values of the macros of ot-br-posix. This is not an exhaustive
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**